redun/executors/gcp_batch.py

```diff
@@ -0,0 +1,584 @@
+                )
+                self.pending_batch_tasks[existing_task.name] = job
+        if batch_task_name is None:
+            self.arrayer.add_job(job)
+
+        self._start()
+
+    def _submit_jobs(self, jobs: List[RedunJob]) -> None:
+        """
+        Callback for JobArrayer to return arrays of Jobs.
+        """
+        if len(jobs) == 1:
+            # Singleton jobs are given as single element lists.
+            self._submit_single_job(jobs[0])
+        else:
+            self._submit_array_job(jobs)
+
+    def _submit_array_job(self, jobs: List[RedunJob]) -> str:
+        """
+        Submits an array job, returning job name uuid
+        """
+        array_size = len(jobs)
+        all_args = []
+        all_kwargs = []
+        for job in jobs:
+            assert job.args
+            all_args.append(job.args[0])
+            all_kwargs.append(job.args[1])
+
+        # All jobs identical so just grab the first one
+        job = jobs[0]
+        assert job.task
+        if job.task.script:
+            raise NotImplementedError("Array jobs not supported for scripts")
+
+        # Generate a unique name for job with no '-' to simplify job name parsing.
+        array_uuid = str(uuid.uuid4())
+
+        task_options = self._get_job_options(job, array_uuid=array_uuid)
+
+        # Setup input, output and error path files. Input file is a pickled list
+        # of args, and kwargs, for each child job.
+        input_file = get_array_scratch_file(self.gcs_scratch_prefix, array_uuid, SCRATCH_INPUT)
+        with File(input_file).open("wb") as out:
+            pickle_dump([all_args, all_kwargs], out)
+
+        # Output file is a plaintext list of output paths, for each child job.
+        output_file = get_array_scratch_file(self.gcs_scratch_prefix, array_uuid, SCRATCH_OUTPUT)
+        output_paths = [
+            get_job_scratch_file(self.gcs_scratch_prefix, job, SCRATCH_OUTPUT) for job in jobs
+        ]
+        with File(output_file).open("w") as ofile:
+            json.dump(output_paths, ofile)
+
+        # Error file is a plaintext list of error paths, one for each child job.
+        error_file = get_array_scratch_file(self.gcs_scratch_prefix, array_uuid, SCRATCH_ERROR)
+        error_paths = [
+            get_job_scratch_file(self.gcs_scratch_prefix, job, SCRATCH_ERROR) for job in jobs
+        ]
+        with File(error_file).open("w") as efile:
+            json.dump(error_paths, efile)
+
+        # Eval hash file is plaintext hashes of child jobs for matching for job
+        # reuniting.
+        eval_file = get_array_scratch_file(self.gcs_scratch_prefix, array_uuid, SCRATCH_HASHES)
+        with File(eval_file).open("w") as eval_f:
+            eval_f.write("\n".join([job.eval_hash for job in jobs]))  # type: ignore
+
+        image = task_options.pop("image", self.image)
+        project = task_options.pop("project", self.project)
+        region = task_options.pop("region", self.region)
+
```

Comment:
I think we need to drop `task_counts` if its present in `task_options` since we explicitly pass it below to `batch_submit()`:

```suggestion

        # Drop task_count, since we can infer it from the number of jobs.
        task_options.pop("task_count", None)
```

I am currently getting this error:

```
Traceback (most recent call last):
  File "/Users/rasmus/projects/redun-dev2/.venv/bin/redun", line 10, in <module>
    client.execute()
  File "/Users/rasmus/projects/redun-dev2/redun/cli.py", line 1040, in execute
    return args.func(args, extra_args, argv)
  File "/Users/rasmus/projects/redun-dev2/redun/cli.py", line 1676, in run_command
    result = scheduler.run(
  File "/Users/rasmus/projects/redun-dev2/redun/scheduler.py", line 947, in run
    raise result.error
  File "/Users/rasmus/projects/redun-dev2/redun/job_array.py", line 108, in _monitor_stale_jobs
    self.submit_pending_jobs(descr)
  File "/Users/rasmus/projects/redun-dev2/redun/job_array.py", line 182, in submit_pending_jobs
    self._submit_jobs(jobs)
  File "/Users/rasmus/projects/redun-dev2/redun/executors/gcp_batch.py", line 296, in _submit_jobs
    self._submit_array_job(jobs)
  File "/Users/rasmus/projects/redun-dev2/redun/executors/gcp_batch.py", line 357, in _submit_array_job
    self.gcs_scratch_prefix, job, job.task, code_file=self.code_file, array_uuid=array_uuid
TypeError: redun.executors.gcp_utils.batch_submit() got multiple values for keyword argument 'task_count'
```

Reply:
Updated with this change.